src/utils/perfagg.py

```diff
@@ -86,6 +87,133 @@
 }
 
 
+# joins disparate runs less dumbly than rocprof
+def join_prof(workload_dir, join_type, out=None):
+    # Set default output directory if not specified
+    if out == None:
+        out = workload_dir + "/pmc_perf.csv"
+    files = glob.glob(workload_dir + "/" + "pmc_perf_*.csv")
+    df = None
+
+    for i, file in enumerate(files):
+        # _df = parse_rocprof_kernels(file)
+
+        _df = pd.read_csv(file)
+        key = _df.groupby("KernelName").cumcount()
+        if join_type == "kernel":
+            _df["key"] = _df.KernelName + " - " + key.astype(str)
+        elif join_type == "grid":
+            _df["key"] = (
+                _df.KernelName + " - " + key.astype(str) + " - " + _df.grd.astype(str)
+            )
+        else:
+            print("ERROR: Unrecognized --join-type")
+            sys.exit(1)
+
+        if df is None:
+            df = _df
+        else:
+            # join by unique index of kernel
+            df = pd.merge(df, _df, how="inner", on="key", suffixes=("", f"_{i}"))
+    # now, we can:
+    #   A) throw away any of the "boring" duplicats
+    df = df[
+        [
+            k
+            for k in df.keys()
+            if not any(
+                check in k
+                for check in [
+                    # "gpu",
+                    "queue-id",
+                    "queue-index",
+                    "pid",
+                    "tid",
+                    # "grd",
```

Comment:
Do we do any checking that e.g., the vgpr counts match between joins?
That might be a good warning to throw (e.g., check if we get any of the suffixes above, and if so warn the user they may want to use a different strategy)

Reply:
https://github.com/AMDResearch/omniperf/blob/44edef4635c18ee3f2bb764f7078db05dfbbecd6/src/utils/perfagg.py#L121-L140